### PR TITLE
Add details & tooltip

### PR DESCRIPTION
**Pull request by krystian50**

### CODE REVIEW

File: src/components/node-details/index.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+// NodeDetails component
+const NodeDetails: React.FC<NodeDetailsProps> = ({ node, onClose }) => {
+  return (
+    <div className="p-4 border border-gray-300 rounded-md shadow-md w-full relative">
+      <button className="absolute top-2 right-2" onClick={onClose}>
+        <XIcon className="h-5 w-5 text-gray-500 hover:text-gray-700" />
+      </button>
+      <div className="mb-2">
+        <span className="font-bold">Label:</span> {node?.name}
```
Comment (tomusdrw) [inline src/components/node-details/index.tsx:19]:
```suggestion
        <span className="font-bold">Node Hash:</span> {node?.name}
```
It's not a label, it's a node hash.

File: src/components/node-details/index.tsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        </div>
+      )}
+      {node?.attributes?.valueHash && (
+        <div className="mb-2">
+          <span className="font-bold">Value Hash:</span> {node?.attributes?.valueHash}
+        </div>
+      )}
+      {node?.attributes?.keyHash && (
+        <div className="mb-2">
+          <span className="font-bold">Key Hash:</span> {node?.attributes?.keyHash}
```
Comment (tomusdrw) [inline src/components/node-details/index.tsx:33]:
```suggestion
          <span className="font-bold">Key:</span> {node?.attributes?.keyHash}
```
Just key, not key hash.

File: src/components/trie/tooltip.tsx [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import { Node } from "@/types/tree";
+
+export const TooltipContent = ({ label, value, valueHash, keyHash }: Node) => {
+  return (
+    <div>
+      <div>
+        <strong>Label:</strong> {label}
```
Comment (tomusdrw) [inline src/components/trie/tooltip.tsx:7]:
```suggestion
        <strong>Hash:</strong> {label}
```

File: src/components/trie/utils.ts [hunk 4]
```diff
@@ -52,7 +52,7 @@ export function trieToTreeUI(
   return {
     name: hash.toString(),
     attributes: {
-      key: leaf.getKey().toString(),
+      keyHash: leaf.getKey().toString(),
```
Comment (tomusdrw) [inline src/components/trie/utils.ts:55]:
```suggestion
      key: leaf.getKey().toString(),
```

Key is fine. I know it "looks" like a hash, but calling it "key hash" implies that the "key" data is hashed again, and that's not correct.

File: src/components/trie/index.tsx [hunk 5]
```diff
@@ -143,17 +154,23 @@ const Trie: React.FC<GraphComponentProps> = ({ treeData }) => {
...
+              keyHash={node.data("keyHash")}
+            />,
+          ),
         });
 
         // Store the tippy instance for cleanup later
         tippyInstances.push(tip);
 
+        node.on("tap", () => {
+          onNodeSelect(node.data("label"));
```
Comment (tomusdrw) [inline src/components/trie/index.tsx:171]: `onNodeSelect` should added as a dep as the warning suggests. Although if you do that it's going to be re-created every time, becuase `onNodeSelect` is not a `useCallback` afaict.

File: src/pages/trie/TriePage.tsx [hunk 6]
```diff
@@ -66,9 +84,17 @@ export const TriePage = () => {
     }
   };
 
+  const onNodeSelect = (node: string) => {
```
Comment (tomusdrw) [inline src/pages/trie/TriePage.tsx:87]: Probably should be `useCallback` to avoid re-rendering.

**Review by tomusdrw — APPROVED** [review]
lgtm! Some small issues with the correctness of the displayed data.

File: src/pages/trie/TriePage.tsx (hunk 6 above)
Comment (krystian50) [reply src/pages/trie/TriePage.tsx:87]: true

File: src/components/trie/index.tsx (hunk 5 above)
Comment (krystian50) [reply src/components/trie/index.tsx:171]: true

**Pull request merged**